### PR TITLE
Homework Completed.

### PR DESCRIPTION
**Pull request by ankeetshankar**
Please find attached the homework for Friday's class.

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -11,19 +11,88 @@
...
 
 ################################################################################
 def main():
+	print("Welcome to the Number guessing game developed in Python by Ankeet")
+	print("\n")
+	print("*"*70)
+	print("The computer has generated a random number, the Player 1 has to guess the number")
+	print("\n")
+	import random
+	number = random.randint(1,26)
```
Comment (danielsgriffin) [inline D04_ex00.py]: I think I mislead you on this. randint is inclusive so you don't need 26.
but please add space

File: D04_ex00.py [hunk 2]
```diff
@@ -11,19 +11,88 @@
...
+		except:
+			print ("Please enter only numbers, integers only. This will count as a valid attempt. Better luck in the next attempt")
+			print("\n")
+			user_tries= user_tries +1
+			continue
+	if user_tries >5:
+		print ("Thanks for playing the game. You were not able to guess the number and have exhausted the attempts")
+		print ("\n")
+		print ("*"*70)
+		print ("/n")
```
Comment (danielsgriffin) [inline D04_ex00.py]: What does this do?

File: D04_ex00.py [hunk 3]
```diff
@@ -11,19 +11,88 @@
 #         - then ends the program
 ################################################################################
 # Imports
-
-
+import random
 # Body
+def check_tries(a):
```
Comment (danielsgriffin) [inline D04_ex00.py]: I think this could have used a bit more testing. Ask me if you can't find some bugs.

File: HW04_ch07_ex02.py [hunk 4]
```diff
@@ -21,15 +21,39 @@
...
+			print ("Thanks for using the eval operator loop. The eval operator was not used a single time")
+			break
+		if expression_user == "done" and counter != 0:
+			print ("Thanks for using the eval operator loop. The last expression being evaluated by the operator is", z )
+			break
+		print (eval (expression_user))
+		z = (eval (expression_user))
+		print ("\n")
+		counter = counter + 1
+		continue
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: Why continue here?

File: HW04_ch07_ex02.py [hunk 5]
```diff
@@ -21,15 +21,39 @@
...
+		if expression_user == "done" and counter == 0:
+			print ("Thanks for using the eval operator loop. The eval operator was not used a single time")
+			break
+		if expression_user == "done" and counter != 0:
+			print ("Thanks for using the eval operator loop. The last expression being evaluated by the operator is", z )
+			break
+		print (eval (expression_user))
+		z = (eval (expression_user))
+		print ("\n")
+		counter = counter + 1
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: += 1

File: HW04_ch07_ex02.py [hunk 6]
```diff
@@ -21,15 +21,39 @@
...
+		expression_user = input ("Please enter the expression to be evaluated in the eval function")
+		
+		if expression_user == "done" and counter == 0:
+			print ("Thanks for using the eval operator loop. The eval operator was not used a single time")
+			break
+		if expression_user == "done" and counter != 0:
+			print ("Thanks for using the eval operator loop. The last expression being evaluated by the operator is", z )
+			break
+		print (eval (expression_user))
+		z = (eval (expression_user))
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: why evaluate twice?

File: HW04_ch08_ex00.py [hunk 7]
```diff
@@ -20,14 +20,30 @@
 
 
 # Body
+def count(string, letter):
+	word = string
+	counter = 0
+	for letter1 in word:
+		if letter1 == letter:
+			counter = counter +1
```
Comment (danielsgriffin) [inline HW04_ch08_ex00.py]: += 1

File: HW04_ch08_ex00.py [hunk 8]
```diff
@@ -20,14 +20,30 @@
...
 
+	print (counter)
```
Comment (danielsgriffin) [inline HW04_ch08_ex00.py]: no space

File: HW04_ch08_ex04.py [hunk 9]
```diff
@@ -43,6 +49,10 @@ def any_lowercase3(s):
 
 def any_lowercase4(s):
     """Explain what is wrong, if anything, here.
+    This function is incorrect as the False condition is or'd with the True or False condition. 
+    This Or statement is in the loop and the last character will be Or'd and Flag will contain the result only for the last character not the entire string 
+    Hence this function is incorrect.
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: Can you show me an example?

File: HW04_ch08_ex05.py [hunk 10]
```diff
@@ -2,3 +2,50 @@
...
+#>>>ord('c') - ord('a')
+#2
+#Because 'c' is the two-eth letter of the alphabet. But beware: the numeric codes for upper case
+#letters are different.
+#Potentially offensive jokes on the Internet are sometimes encoded in ROT13, which is a Caesar
+#cypher with rotation 13. If you are not easily offended, find and decode some of them. Solution:
+#http: // thinkpython2. com/ code/ rotate. py .
+#
+
+def rotate_word(string,rotator_number):
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: def rotate_word(string, rotator_number):

File: HW04_ch08_ex05.py [hunk 11]
```diff
@@ -2,3 +2,50 @@
...
+	z =""
+	#This function takes in two parameters string and rotator_number.
+	for x in range(0,len(string)):
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: +	for x in range(0, len(string)):

File: HW04_ch08_ex05.py [hunk 12]
```diff
@@ -2,3 +2,50 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]:
```
Book Examples:
 OK  'A'     by 3   => got: 'D'     expected: 'D'
  X  'Z'     by 1   => got: '['     expected: 'A'
 OK  'cheer' by 7   => got: 'jolly' expected: 'jolly'
  X  'melon' by -10 => got: 'c[bed' expected: 'cubed'
 OK  'IBM'   by -1  => got: 'HAL'   expected: 'HAL'

Daniel's Tests:
 OK  'abc'   by 1   => got: 'bcd'   expected: 'bcd'
  X  'xyz'   by 1   => got: 'yz{'   expected: 'yza'
 OK  'ABC'   by 1   => got: 'BCD'   expected: 'BCD'
  X  'XYZ'   by 1   => got: 'YZ['   expected: 'YZA'
  X  'Az'    by 1   => got: 'B{'    expected: 'Ba'
  X  'Aa'    by 26  => got: '[{'    expected: 'Aa'
  X  'abc'   by -1  => got: '`ab'   expected: 'zab'
 OK  'xyz'   by -1  => got: 'wxy'   expected: 'wxy'
  X  'ABC'   by -1  => got: '@AB'   expected: 'ZAB'
 OK  'XYZ'   by -1  => got: 'WXY'   expected: 'WXY'
  X  'Az'    by -1  => got: '@y'    expected: 'Zy'
  X  'Aa'    by -26 => got: "'G"    expected: 'Aa'
```

**Comment by danielsgriffin** [comment]
Good job. Make sure you test your code thoroughly.

File: HW04_string2.py [hunk 13]
```diff
@@ -18,9 +18,23 @@
 # Return the resulting string.
 def verbing(s):
     # +++your code here+++
-    return
```
Comment (nyhyang) [inline HW04_string2.py]: I think this is an extra return.

File: HW04_string2.py [hunk 14]
```diff
@@ -42,8 +67,27 @@ def not_bad(s):
 # Given 2 strings, a and b, return a string of the form
 #  a-front + b-front + a-back + b-back
 def front_back(a, b):
+    if (len(a)%2) == 0:
```
Comment (nyhyang) [inline HW04_string2.py]: I would add spaces before and after % and I don't thin you need the parenthesis 
`if len(a) % 2 == 0:`

File: HW04_string2.py [hunk 15]
```diff
@@ -42,8 +67,27 @@ def not_bad(s):
...
+        a_front = a[0:(len(a))//2]
+        a_back = a[((len(a))//2):]
+
+    else:
+        a_front = a[0:((len(a)//2)+1)]
+        a_back = a[((len(a)//2)):]
+
+    if (len(b)%2) == 0:
```
Comment (nyhyang) [inline HW04_string2.py]: same as the top comment

**Pull request closed without merging**